tfkbnufft/mri/dcomp_calc.py

```diff
@@ -82,3 +83,35 @@ def calculate_radial_dcomp_tf(interpob, nufftob_forw, nufftob_back, ktraj, stack
     )
 
     return dcomp
+
+
+def calculate_density_compensator(interpob, ktraj, num_iterations=10):
+    """Numerical density compensation estimation for a any trajectory.
+
+    Estimates the density compensation function numerically using a NUFFT
+    interpolator operator and a k-space trajectory (ktraj).
+    This function implements Pipe et al
+
+    This function uses a nufft hyper parameter dictionary, the associated nufft
+    operators and k-space trajectory.
+
+    Args:
+        interpob (dict): the output of `KbNufftModule._extract_nufft_interpob`
+            containing all the hyper-parameters for the nufft computation.
+        ktraj (tensor): The k-space trajectory in radians/voxel dimension (d, m).
+            d is the number of spatial dimensions, and m is the length of the
+            trajectory.
+        num_iterations (int): default 10
+            number of iterations
+
+    Returns:
+        tensor: The density compensation coefficients for ktraj of size (m).
+    """
+    test_sig = tf.ones([1, 1, ktraj.shape[1]], dtype=tf.complex64)
+    for i in range(num_iterations):
+        test_sig = test_sig / tf.cast(tf.math.abs(kbinterp(
+            adjkbinterp(test_sig, ktraj[None, :], interpob),
+            ktraj[None, :],
+            interpob
+        )), 'complex64')
```

Comment:
it could be nice to make sure that the returned value has the same dimensionality as `calculate_radial_dcomp_tf`.
```suggestion
        )), 'complex64')
    test_sig = test_sig[0, 0]
```
it would also correspond to the doc